src/Admin/BaseFieldDescription.php

```diff
@@ -413,4 +428,56 @@ public function isVirtual()
     {
         return false !== $this->getOption('virtual_field', false);
     }
+
+    private function getFieldGetterKey($object, $fieldName)
+    {
+        if (!is_string($fieldName)) {
+            return null;
+        }
+        $components = [get_class($object), $fieldName];
```

Comment:
@andrewmy, I had to replace this line with `$components = [$object ? get_class($object) : null, $fieldName];`, since passed `$object` can be `null`. Could you please check this in order to confirm it is a valid fix?

Reply:
I think that if `$object` is `null` then `getFieldValue` should either return `null` or throw an exception. You shouldn't try to get a field from null, right?

Reply:
I haven't analyzed the case yet, but it's petty possible. I'll ping you if I can confirm that. But my point here is that there is no check in previous code to avoid this, so there is a warning now, given the passed `null` value.

Reply:
@andrewmy, confirmed. I'm sending `null` to `BaseFieldDescription::getFieldValue()`. The issue here seems to be although `FieldDescriptionInterface` defines the argument 1 as `object` in the docblock, there is no additional check in the implementation, so it never raised any error or thrown any exception previous to your commit; so I don't know if there can be addressed as a BC break or if the argument check should be added at the mentioned method.

**Context:**
As context, I must mention the scenario in which the received subject is `null`.
I have a simple set of geo entities (**City**, **State**, **Country**) which have many-to-one association respectively (a country may have _n_ states, a state may have _n_ cities). I use these entities within others, like **User**. The association between **User** and **City** is many-to-one and is **NULLABLE**. So, in my `UserAdmin::configureShowFields()` I have the following call:
```php
ShowMapper::add('city.state', EntityType::class, ['class' => State::class])
```
then, when there is no city for a user, value passed to `BaseFieldDescription::getFieldValue()` is null.

/cc @greg0ire.

Reply:
This is a valid concern and would break before this patch too. I think it's ground for a new PR which would handle null field values. By returning `null` too, I guess?

Reply:
Do you mean something like this?
```php
if (!is_string($fieldName) || null === $object) {
    return null;
}
```